app/validation/validator.py

```diff
@@ -348,39 +353,77 @@ def _validate_skip_condition(self, skip_condition, answer_ids_with_group_id, blo
         errors.extend(self._validate_when_rule(when, answer_ids_with_group_id, block_or_group['id']))
         return errors
 
-    def _validate_list_collector_type(self, block):  # noqa: C901  pylint: disable=too-complex
+    def _validate_list_collector_type(self, block):  # noqa: C901  pylint: disable=too-complex, too-many-locals
         errors = []
-        questions = []
-        add_answer_value = block['add_answer_value']
-
-        if 'question' in block:
-            questions.append(block['question'])
-
-        if 'question_variants' in block:
-            for question in block['question_variants']:
-                questions.append(question)
-
-        for question in questions:
-            for answer in question['answers']:
-                if answer['type'] != 'Radio':
+        collector_questions = self._get_all_questions_for_block(block)
+        remove_questions = self._get_all_questions_for_block(block['remove_block'])
+        add_answer_value = block['add_answer']['value']
```

Comment:
Need to also validate that the `block['add_answer']['id']` is an `id` that is present within the block as runner will 500 with a KeyError if they mismatch.